shim.js

```diff
@@ -26,3 +26,21 @@ if (typeof localStorage !== 'undefined') {
 // If using the crypto shim, uncomment the following line to ensure
 // crypto is loaded first, so it can populate global.crypto
 require('crypto')
+
+const TextEncoder = require('text-encoding').TextEncoder;
+const TextDecoder = require('text-encoding').TextDecoder;
+
+global.TextDecoder = TextDecoder;
+global.TextEncoder = TextEncoder;
+
+if (typeof btoa === 'undefined') {
+  global.btoa = function (str) {
+    return Buffer.from(str, 'binary').toString('base64');
+  };
+}
+
+if (typeof atob === 'undefined') {
+  global.atob = function (b64Encoded) {
+    return Buffer.from(b64Encoded, 'base64').toString('binary');
+  };
+}
```

Comment:
shims needed by wallet-connect

Reply:
Added a simple comment indicating that those shims are needed by wallet-connect

6471cb635e1a93b38bba4c0ac5e2109e543e90bb